test/arguments/escape-no-icu.js

```diff
@@ -0,0 +1,16 @@
+// Mimics Node.js when built without ICU support
+// See https://github.com/sindresorhus/execa/issues/1143
+globalThis.RegExp = class extends RegExp {
+	constructor(regExpString, flags) {
+		if (flags?.includes('u') && regExpString.includes('\\p{')) {
+			throw new Error('Invalid property name');
+		}
+
+		super(regExpString, flags);
+	}
+
+	static isMocked = true;
+};
+
+// Execa computes the RegExp when first loaded, so we must delay this import
+await import('./escape.js');
```

Comment:
We could mock/unmock `RegExp` directly in `escape.js`, and run each test twice. But then each test would need to be serial. Also it makes the test file a little complex (this was my first approach). Using a separate test file that imports the other one results in less code.